src/views/Dashboard/sidebar/index.js

```diff
@@ -41,26 +41,27 @@ export class Sidebar extends Component {
   render() {
     let backdrop;
     const { sideDrawerOpen } = this.state;
-    const { children } = this.props;
+    const { children, profileError, profile } = this.props;
+    const { tokenValid } = this.state;
+
     if (sideDrawerOpen) {
       backdrop = <Backdrop click={this.backdropClickHandler} />;
     }
 
-    if(this.state.tokenValid) {
-      if (this.props.profileError) {        
-        if(this.props.profileError.error) {
-          return <Redirect to="/verify-email"/>;
-        }
-        if(this.props.profileError.message) {
-          localStorage.removeItem('token')
-          return <Redirect to="/signin"/>;
-        }
-      }}
-    if (this.props.profile.data) {
-      if (!this.props.profile.data.managerId) {
-        return <Redirect to="/profile"/>;
-      }
+ 
+    if(tokenValid && profileError && profileError.error) {
+      return <Redirect to="/verify-email"/>;
+    }
```

Comment:
@NiyoEric , What if you try something like this.
```
tokenValid && profileError && profileError.error && <Redirect to="/verify-email"/>
```

Reply:
@hezronkimutai, I have tried it out but it blocks `Redirect` functionality.
> Code used
``` jsx
   tokenValid && profileError && profileError.error && <Redirect to="/verify-email"/>;
```
*Output*
<img width="1440" alt="Screen Shot 2020-02-21 at 11 03 49" src="https://user-images.githubusercontent.com/33821734/75019669-fb3c8b80-5499-11ea-9f0e-523ad88b5b1c.png">